samples/dnn/text_detection.cpp

```diff
@@ -159,3 +220,71 @@ void decode(const Mat& scores, const Mat& geometry, float scoreThresh,
         }
     }
 }
+
+void four_points_transform(const Mat& frame, const RotatedRect& box, const Size2f& box_size, const Point2f& ratio,
+                           Mat& result) {
+
+    Point2f vertices[4];
+    box.points(vertices);
+    for (int j = 0; j < 4; ++j)
+    {
+        vertices[j].x *= ratio.x;
+        vertices[j].y *= ratio.y;
+    }
+
+
+    Point2f target_vertices[4] = {Point2f(0, box_size.height - 1),
+                                  Point2f(0, 0), Point2f(box_size.width - 1, 0),
+                                  Point2f(box_size.width - 1, box_size.height - 1),
+                                  };
+    // Compute width
+    int width_a = std::sqrt(std::pow(target_vertices[1].x - target_vertices[2].x, 2)
+                + std::pow(target_vertices[2].y - target_vertices[3].y, 2));
+
+    int width_b = std::sqrt(std::pow(target_vertices[0].x - target_vertices[3].x, 2)
+                + std::pow(target_vertices[0].y - target_vertices[3].y, 2));
+
+    int max_width = std::max(width_a, width_b);
+
+    // Compute height
+    int height_a = std::sqrt(std::pow(target_vertices[0].x - target_vertices[1].x, 2)
+                + std::pow(target_vertices[0].y - target_vertices[1].y, 2));
+
+    int height_b = std::sqrt(std::pow(target_vertices[2].x - target_vertices[3].x, 2)
+                + std::pow(target_vertices[2].y - target_vertices[3].y, 2));
+
+    int max_height = std::max(height_a, height_b);
+
+    Mat rotation_matrix = getPerspectiveTransform(vertices, target_vertices);
+    warpPerspective(frame, result, rotation_matrix, Size(max_width, max_height));
```

Comment:
`dsize` should be a size of destination image - 100x32

Reply:
@rngtna, we do `warpPerspective` and then resize by `blobFromImage`. Why not to warp to the final 100x32 size?